cli/config/config.go

```diff
@@ -126,11 +126,38 @@ func NewJobConfiguration(cfgFilePath string) (JobConfiguration, error) {
 	return c, nil
 }
 
+type SpecConfig struct {
+	Specs []string `yaml:"specs,omitempty"`
+}
+
+func NewSpecConfig(files []string) SpecConfig {
+	specConfig := SpecConfig{}
+	specConfig.Specs = append(specConfig.Specs, files...)
+	return specConfig
+}
+
+const SPEC_CONFIG_TEMP_PREFIX = "spec-config"
+const SPEC_CONFIG_FILENAME = "specConfig.yaml"
+
+func NewSpecConfigFile(files []string, filepath string) (error) {
```

Comment:
Totally optional, but one way of doing this is having the return value be `err error` and then you don't need to say `return err` just simply `return` will return the value assigned to `err`

Reply:
That's actually not recommended by default, but can be used when you have named return values.

Reply:
Apparently this practice is discouraged. Disregard ^

Reply:
yeah, this is something we can agree as good practices I usually like to always be explicit